requester/requester.go

```diff
@@ -310,8 +310,7 @@ func (b *Work) runWorker(parent context.Context, client *http.Client, n int) {
 			return http.ErrUseLastResponse
 		}
 	}
-
-	ctx, cancel := context.WithCancel(parent)
+	ctx, cancel := context.WithTimeout(context.Background(), b.Req.clientTimeout())
```

Comment:
If you are creating a new context for every worker, you lose the connection to the `RunTimeout`. What is the motivation for this change? I think we should keep passing in the parent context, so the context is cancelled when the run timeout is reached.

Reply:
You're right that the call to `cancel()` at https://github.com/elastic/hey-apm/pull/63/files#diff-c834c625eb05c1eb841e029b03e3c7bcR317 triggers the cancellation of the child context. 

However, I still think you should keep calling the `cancel` function for the `makeRequest` https://github.com/elastic/hey-apm/pull/63/files#diff-c834c625eb05c1eb841e029b03e3c7bcR89. Several requests can be made within one load test run. The contexts for those requests do not get cancelled, only the context of the last request gets cancelled when the time is up or on an interrupt.

Reply:
> The contexts for those requests do not get cancelled, only the context of the last request gets cancelled when the time is up or on an interrupt

How so?

Reply:
after some offline discussion - go ahead with the contexts as they are.